photon-core/src/main/java/org/photonvision/common/hardware/HardwareManager.java

```diff
@@ -160,21 +165,60 @@ public boolean restartDevice() {
         }
     }
 
-    public void setStatus(ProgramStatus status) {
-        switch (status) {
-            case UHOH:
-                // red flashing, green off
-                break;
-            case RUNNING:
-                // red solid, green off
-                break;
-            case RUNNING_NT:
-                // red off, green solid
-                break;
-            case RUNNING_NT_TARGET:
-                // red off, green flashing
-                break;
+    public void setNTConnected(boolean isConnected) {
+        this.ntConnected = isConnected;
+    }
+
+    public void setRunning(boolean isRunning) {
+        this.systemRunning = isRunning;
+    }
+
+    public void statusLEDUpdate() {
+        // make blinky
+        boolean blinky = ((blinkCounter % 2) == 0);
+
+        // check if any pipeline has a visible target
+        boolean anyTarget = false;
+        for (var t : this.pipelineTargets) {
+            if (t) {
+                anyTarget = true;
+            }
+        }
+
+        if (this.systemRunning) {
+            if (!this.ntConnected) {
+                if (anyTarget) {
+                    // alternate blue/yellow
+                    statusLED.setRGB(blinky, blinky, !blinky);
+                } else {
+                    // Yellow flashing
+                    statusLED.setRGB(blinky, blinky, false);
+                }
+            } else {
+                if (anyTarget) {
+                    // alternate blue/green
+                    statusLED.setRGB(false, blinky, !blinky);
+                } else {
+                    // blinky green
+                    statusLED.setRGB(false, blinky, false);
+                }
+            }
+        } else {
+            // Faulted, not running... blinky red
+            statusLED.setRGB(blinky, false, false);
         }
+
+        blinkCounter++;
+    }
+
+    private ArrayList<Boolean> pipelineTargets = new ArrayList<Boolean>();
+    private boolean ntConnected = false;
+    private boolean systemRunning = false;
+    int blinkCounter = 0;
+
+    public void setTargetsVisibleStatus(int pipelineIdx, boolean hasTargets) {
+        pipelineTargets.ensureCapacity(pipelineIdx);
```

Comment:
Isn't cameraIdx more accurate? Also, a map might be a better data structure here

Reply:
I like how I had amnesia that maps existed last night. Yes, map 1000% better.